test/integration/apiserver/apply/reset_fields_test.go

```diff
@@ -298,3 +297,31 @@ func TestApplyResetFields(t *testing.T) {
 		}
 	}
 }
+
+func expectConflict(t *testing.T, objRet *unstructured.Unstructured, err error, dynamicClient dynamic.Interface, resource schema.GroupVersionResource, namespace, name, where string) {
+	if err != nil && strings.Contains(err.Error(), "conflict") {
+		return
+	}
+	which := "returned"
+	var gotten string
+	var err2 error
+	if objRet == nil {
+		which = "subsequently fetched"
+		objRet, err2 = dynamicClient.
+			Resource(resource).
+			Namespace(namespace).
+			Get(context.TODO(), name, metav1.GetOptions{})
+		if err2 != nil {
+			gotten = fmt.Sprintf("<failed to Get object: %v>", err2)
```

Comment:
You should do `t.Fatal` here (or better, return the error) with the right string, and I think if you do that you can get rid of both `which` and `gotten`.

Reply:
Revised as suggested.